src/server/cmd/pachctl/cmd/cmd_test.go

```diff
@@ -2,24 +2,57 @@ package cmd
 
 import (
 	"encoding/json"
+	"fmt"
 	"io"
 	"os"
 	"testing"
 
+	"github.com/pachyderm/pachyderm/src/client/pkg/config"
 	"github.com/pachyderm/pachyderm/src/client/pkg/require"
 	deploycmds "github.com/pachyderm/pachyderm/src/server/pkg/deploy/cmds"
 	api "k8s.io/kubernetes/pkg/api/v1"
 )
 
-func TestMetrics(t *testing.T) {
+func TestMetricsNormalDeployment(t *testing.T) {
+	// Run deploy normally, should see METRICS=true
+	testDeploy(t, false, false, true)
+}
 
+func TestMetricsNormalDeploymentNoMetricsFlagSet(t *testing.T) {
 	// Run deploy normally, should see METRICS=true
+	testDeploy(t, false, true, false)
+}
+
+func TestMetricsDevDeployment(t *testing.T) {
+	// Run deploy w dev flag, should see METRICS=false
+	testDeploy(t, true, false, false)
+}
+
+func TestMetricsDevDeploymentNoMetricsFlagSet(t *testing.T) {
+	// Run deploy w dev flag, should see METRICS=false
+	testDeploy(t, true, true, false)
+}
+
+func testDeploy(t *testing.T, devFlag bool, noMetrics bool, expectedEnvValue bool) {
```

Comment:
Yeah, your understanding is correct it won't run in parallel as is. This is just a defensive thing in case someone comes along and makes the tests parallel.